src/pthread_esm_startup.mjs

```diff
@@ -15,14 +15,17 @@ console.log("Running pthread_esm_startup");
 
 #if ENVIRONMENT_MAY_BE_NODE
 // Create as web-worker-like an environment as we can.
-var worker_threads = await import('worker_threads');
-global.Worker = worker_threads.Worker;
-var parentPort = worker_threads['parentPort'];
-parentPort.on('message', (msg) => global.onmessage?.({ data: msg }));
-Object.assign(globalThis, {
-  self: global,
-  postMessage: (msg) => parentPort['postMessage'](msg),
-});
+globalThis.self = globalThis;
```

Comment:
I noticed that code here was lacking a check for node.  Added in: https://github.com/emscripten-core/emscripten/pull/25948